src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java

```diff
@@ -1,33 +1,49 @@
 package ee.tuleva.onboarding.auth.session;
 
-import java.io.Serial;
 import java.io.Serializable;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
+import javax.servlet.http.HttpSession;
 import org.springframework.stereotype.Component;
-import org.springframework.web.context.annotation.SessionScope;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
 
 @Component
-@SessionScope
-public class GenericSessionStore implements Serializable {
+public class GenericSessionStore {
```

Comment:
What changed here? Not a session scoped bean anymore?

Reply:
Yes, injecting a session scoped bean will create a session automatically which we do not want